src/hip_memory.cpp

```diff
@@ -1861,182 +1932,78 @@ hipError_t hipMemcpyParam2DAsync(const hip_Memcpy2D* pCopy, hipStream_t stream)
 // TODO-sync: function is async unless target is pinned host memory - then these are fully sync.
 hipError_t hipMemsetAsync(void* dst, int value, size_t sizeBytes, hipStream_t stream) {
     HIP_INIT_SPECIAL_API(hipMemsetAsync, (TRACE_MCMD), dst, value, sizeBytes, stream);
-
-    hipError_t e = hipSuccess;
-
     stream = ihipSyncAndResolveStream(stream);
-
-    e = ihipMemset(dst, value, sizeBytes, stream, ihipMemsetDataTypeChar);
-
-    return ihipLogStatus(e);
-};
+    return ihipLogStatus(ihipMemsetAsync(dst, value, sizeBytes, stream, ihipMemsetDataTypeChar));
+}
 
 hipError_t hipMemsetD32Async(hipDeviceptr_t dst, int value, size_t count, hipStream_t stream) {
     HIP_INIT_SPECIAL_API(hipMemsetD32Async, (TRACE_MCMD), dst, value, count, stream);
-
-    hipError_t e = hipSuccess;
-
     stream = ihipSyncAndResolveStream(stream);
-
-    e = ihipMemset(dst, value, count, stream, ihipMemsetDataTypeInt);
-
-    return ihipLogStatus(e);
-};
+    return ihipLogStatus(ihipMemsetAsync(dst, value, count, stream, ihipMemsetDataTypeInt));
+}
 
 hipError_t hipMemset(void* dst, int value, size_t sizeBytes) {
     HIP_INIT_SPECIAL_API(hipMemset, (TRACE_MCMD), dst, value, sizeBytes);
-
-    hipError_t e = hipSuccess;
-
-    hipStream_t stream = hipStreamNull;
-    stream =  ihipSyncAndResolveStream(stream);
-    if (stream) {
-        e = ihipMemset(dst, value, sizeBytes, stream, ihipMemsetDataTypeChar);
-        stream->locked_wait();
-    } else {
-        e = hipErrorInvalidValue;
-    }
-    return ihipLogStatus(e);
+    hipStream_t stream = ihipSyncAndResolveStream(hipStreamNull);
```

Comment:
Move stream resolution inside ihipMemset, there is no point in syncing if ihipMemset returns early due to count =0 or !dst.